### PR TITLE
Single product

### PR DESCRIPTION
**Pull request by HarshnaNHaswani**
Please review
deploy preview: https://deploy-preview-19--ecom-meltling-moments-frontend.netlify.app/

### CODE REVIEW

**Review by sri-rishi — APPROVED** [review]
code looks good
good use of semantic tags
feel  free to merge

File: src/router/Router.jsx [hunk 1]
```diff
@@ -28,9 +30,20 @@ export function Router() {
...
+          children: [
+            {
+              path: ":productId",
+              element: <SingleProduct />,
+            },
+            {
+              index: true,
+              element: <ProductCrashFallback />,
+            },
+          ]
```
Comment (dev-enforced) [inline src/router/Router.jsx:36]: Not able to understand this new way of alloting routes.Could you highlight this format?

**Review by anjali1102 — APPROVED** [review]

**Review by dev-enforced — APPROVED** [review]
The live preview is working as expected. Do check out the loading of product page data on refresh as well as for different products itself

File: src/router/Router.jsx (hunk 1 above)
Comment (HarshnaNHaswani) [reply src/router/Router.jsx:36]: Here routes are defined using `useRoutes()` hook
first it enters the path `/product`
if a valid `:productId` is present,
it'll go to `/product/:productId`
else it'll go to the `Fallback`

**Comment by HarshnaNHaswani** [comment]
thank you @anjali1102  @sri-rishi  @dev-enforced  for  pr

**Pull request merged**